### PR TITLE
[CPM-164] search courses regulation modal

### PR DESCRIPTION
**Pull request by Richardng12**
## Changes

- Added search bar to course regulations linked to courses in db
- Added ability to add fake ghost courses 
- Changed workflow to only have 1 textbox

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [x] Can somebody check the styling/CSS?
- Might still need to reorder some things to make workflow nicer

Addresses [CPM-164]

[CPM-164]: https://momentum-agile.atlassian.net/browse/CPM-164

### CODE REVIEW

File: client/src/pages/Courses/RegulationModal.js [hunk 1]
```diff
@@ -77,24 +105,28 @@ const RegulationModal = ({ isOpen, onClose, title, updateCourse, course, editReg
...
+        if (actionMeta.action === "select-option") {
+            const latestCourses = newValue.map((x) => x.label);
+            handleCourseSelect(latestCourses);
         }
 
-        console.log(courseTextValue);
-        updateReg("courses")([...currentReg.courses, courseTextValue], "POINTS");
-        setCourseTextValue("");
+        if (actionMeta.action === "pop-value" || actionMeta.action === "remove-value" || actionMeta.action === "clear") {
+            console.log("remove)");
```
Comment (daltonlim) [inline client/src/pages/Courses/RegulationModal.js:117]: pls remove

**Review by daltonlim — APPROVED** [review]
LGTM

File: client/src/pages/Courses/RegulationModal.js (hunk 1 above)
Comment (Richardng12) [reply client/src/pages/Courses/RegulationModal.js]: done

**Review by FocalChord — APPROVED** [review]
~_^ very nice bretheren

**Pull request merged**